testing/test_subtests.py

```diff
@@ -0,0 +1,840 @@
+from __future__ import annotations
+
+from pathlib import Path
+import sys
+from typing import Literal
+
+import pytest
+
+
+IS_PY311 = sys.version_info[:2] >= (3, 11)
+
+
+@pytest.mark.parametrize("mode", ["normal", "xdist"])
+class TestFixture:
```

Comment:
Would be nice to have a test for the combination of parametrization + subtests. I'm sure it works, but it's the sort of thing that I can see *might* break if a different implementation strategy was used.